subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/worker/DeviceWorker.kt

```diff
@@ -30,18 +31,26 @@ class DeviceWorker(
         Executors.newSingleThreadExecutor().asCoroutineDispatcher()
     )
 
-    private var state: State = device.state()
-
     fun run() = scope.launch {
 
+        var state: State = try {
+            checkDeviceAlive()
+            device.state()
+        } catch (t: Exception) {
+            device.notifyError("Error while checking device initial state", t)
+            // No intention is lost. DeviceWorkerMessage.WorkerFailed event is unnecessary.
+            // Can't use this device any more. ReservationClient will get a new one.
+            return@launch
+        }
+
         for (intention in intentionsRouter.observeIntentions(state)) {
             try {
                 checkDeviceAlive()
```

Comment:
This is ok to check again. The worker will receive an intention later after the initial check.